airbyte-lib/airbyte_lib/factories.py

```diff
@@ -48,20 +48,41 @@ def new_local_cache(
 
 def get_connector(
     name: str,
-    version: str = "latest",
-    config: Optional[Dict[str, Any]] = None,
+    version: str | None = None,
+    pip_url: str | None = None,
```

Comment:
I considered that also, but I think we still need the name separately.

The name gives us a three important things, I think:
1. It tell us the CLI entrypoint so we can invoke the connector's CLI.
2. it tells us how to find the connector in the registry metadata.
3. it tells us what the source should be called when we log about it or store metadata about it

While it's theoretically possible to find the CLI name and connector name, it's somewhat unreliable to do so, and can create more confusion for the user because it relies on somewhat "magic" methods that might not match their expectation. So, for instance, `airbyte-source-mysql`, `pypi-fork-of-source-mysql`, and `git+https://github.com/airbytehq/airbyte.git#subdirectory=integrations/connectors/source-mysql` could all have a name of `source-mysql`, or the forked version might have a different name. The package name might not also match the CLI entrypoint name (`airbyte-source-mysql` vs `source-mysql`), so getting explicit "name" that we also use as the CLI entrypoint just makes the debugging much simpler if there happens to be divergence from the norm.

Admittedly, this metadata may only be able to taken loosely if we're overriding the install source. Sometimes it will still _probably_ apply (like when installing from a feature branch) and sometimes it will not (such when pulling from a completely separate implementation of the connector created by a different author (very rare).

Reply:
sounds fair, I'm not too worried about convenience for this path as it will be somewhat of an expert feature anyway